documentation/src/docs/asciidoc/user-guide/extensions.adoc

```diff
@@ -289,6 +289,20 @@ Specifically, a custom extension can be registered by supplying its fully qualif
 name in a file named `org.junit.jupiter.api.extension.Extension` within the
 `/META-INF/services` folder in its enclosing JAR file.
 
+[[automatic-extension-registration-pattern-matching]]
+===== Pattern Matching Syntax
+
+Pattern matching syntax is used to automatically detect and register extensions. Here are some examples of pattern matching syntax:
+
+- `*`: includes all extensions.
+- `org.junit.*`: includes every extension under the `org.junit` base package and any of its subpackages.
+- `*.MyExtension`: includes every extension whose simple class name is exactly `MyExtension`.
+- `*System*`: includes every extension whose FQCN contains `System`.
+- `*System*, *Dev*`: includes every extension whose FQCN contains `System` or `Dev`.
+- `org.example.MyExtension, org.example.TheirExtension`: includes extensions whose FQCN is exactly `org.example.MyExtension` or `org.example.TheirExtension`.
+
+> **Note**: A class that matches both an inclusion and exclusion pattern will be excluded.
```

Comment:
We should instead just link to the `running-tests-config-params-deactivation-pattern`.

Reply:
I'm not very skilled at documentation work yet 😅

```
[[automatic-extension-registration-pattern-matching]]
===== Pattern Matching Syntax

Refer to <<running-tests-config-params-deactivation-pattern>> for details.
```

Would this change be fine?

Reply:
I made this change in https://github.com/junit-team/junit5/pull/4120/commits/13c1b1986a416b7cf78e18d2efbf1fbdb3bb8f0d